src/java.base/share/classes/javax/crypto/KEM.java

```diff
@@ -222,7 +222,13 @@ public Encapsulated encapsulate() {
          *          to be returned, inclusive
          * @param to the final index of the shared secret byte array
          *          to be returned, exclusive
-         * @param algorithm the algorithm name for the secret key that is returned
+         * @param algorithm the algorithm name for the secret key that is returned.
+         *          See the SecretKey Algorithms section in the
+         *          <a href="{@docRoot}/../specs/security/standard-names.html#secretkey-algorithms">
+         *          Java Security Standard Algorithm Names Specification</a>
+         *          for information about standard secret key algorithm names.
+         *          Use "Generic" if the output will be used as the input keying
+         *          material of a key derivation function (KDF).
```

Comment:
Add link to KDF API?

Reply:
In the original commit at https://github.com/openjdk/jdk/pull/22650/commits/f7c03f781a90ebf3f634c3e7478124463757de17 I used the link. And then, I thought KDF is still not final, and I wanted to convey the idea that it could be accepted by an existing KDF implementation that doesn’t extend `KDFSpi` but is still conceptually a KDF. Perhaps I overthought it?